python/ray/_raylet.pyx

```diff
@@ -1145,3 +1145,48 @@ cdef class CoreWorker:
 
     def current_actor_is_asyncio(self):
         return self.core_worker.get().GetWorkerContext().CurrentActorIsAsync()
+
+    def in_memory_store_get_async(self, object_id, future):
+        cdef:
+            CObjectID c_object_id = (<ObjectID>object_id).native()
+        self.core_worker.get().GetAsync(
+            c_object_id,
+            async_set_result_callback,
+            <void*>future)
+
+cdef void async_set_result_callback(shared_ptr[CRayObject] obj,
+                                    CObjectID object_id,
+                                    void *future) with gil:
+    """This callback is expected to be called from InMemoryStore.
+
+    It wraps the result CRayObject to AsyncGetResponse.
+    """
+    cdef:
+        c_vector[shared_ptr[CRayObject]] objects_to_deserialize
+
+    # void* and cast to object is the Cython recommended way of passing python
+    # object into callback.
+    # https://github.com/cython/cython/blob/master/Demos/callback/cheese.pyx
+    py_future = <object>(future)
+    loop = py_future._loop
+
+    if obj.get().IsInPlasmaError():
```

Comment:
hmmm can we hide this logic in the core worker instead of exposing it here? maybe separate callbacks for in-memory and plasma objects?

Reply:
This callback is only used for in-memory object.